From 48612c018d16b6dae41fb9b4bd6a50ffd2e5cd27 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:31:30 -0400
Subject: [alsa] usb-audio: Prevent printk ratelimiting from spamming kernel log while DEBUG not defined

Message-id: <140972823573-284-patchset-jkysela@redhat.com>
Patchwork-id: 90980
O-Subject: [RHEL 7.1 PATCH 283/442] ALSA: usb-audio: Prevent printk ratelimiting from spamming kernel log while DEBUG not defined
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This (widely used) construction:

if(printk_ratelimit())
dev_dbg()

Causes the ratelimiting to spam the kernel log with the "callbacks suppressed"
message below, even while the dev_dbg it is supposed to rate limit wouldn't
print anything because DEBUG is not defined for this device.

[  533.803964] retire_playback_urb: 852 callbacks suppressed
[  538.807930] retire_playback_urb: 852 callbacks suppressed
[  543.811897] retire_playback_urb: 852 callbacks suppressed
[  548.815745] retire_playback_urb: 852 callbacks suppressed
[  553.819826] retire_playback_urb: 852 callbacks suppressed

So use dev_dbg_ratelimited() instead of this construction.

Signed-off-by: Sander Eikelenboom <linux@eikelenboom.it>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: b7a7723513dc89f83d6df13206df55d4dc26e825)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/pcm.c b/sound/usb/pcm.c
index 131336d..c62a165 100644
--- a/sound/usb/pcm.c
+++ b/sound/usb/pcm.c
@@ -1501,9 +1501,8 @@ static void retire_playback_urb(struct snd_usb_substream *subs,
   * The error should be lower than 2ms since the estimate relies
   * on two reads of a counter updated every ms.
   */
- if (printk_ratelimit() &&
-     abs(est_delay - subs->last_delay) * 1000 > runtime->rate * 2)
-  dev_dbg(&subs->dev->dev,
+ if (abs(est_delay - subs->last_delay) * 1000 > runtime->rate * 2)
+  dev_dbg_ratelimited(&subs->dev->dev,
    "delay: estimated %d, actual %d\n",
    est_delay, subs->last_delay);
 
-- 
1.7.1